test/cases/global_id_test.rb

```diff
@@ -18,6 +18,37 @@ class GlobalIDTest < ActiveSupport::TestCase
       GlobalID.app = nil
     end
   end
+
+  test '.parse returns `nil` for invalid model id' do
+    assert_nil GlobalID.parse('gid://app/Person/1/2')
+    assert_nil GlobalID.parse('gid://app/CompositePrimaryKeyModel/tenant-key-value/id-value/something_else')
+    assert_nil GlobalID.parse('gid://app/CompositePrimaryKeyModel/tenant-key-value/')
+    assert_nil GlobalID.parse('gid://app/CompositePrimaryKeyModel/tenant-key-value')
```

Comment:
I think we are protecting too much. All those glogal ids should be valid. If they can't be located, isn't a job of the parser to find out.

Reply:
>isn't a job of the parser to find out.

I asked in the comment above but would you prefer `Locator` to be the one to check that identifier size matches model's primary key to avoid loading more records than necessary?